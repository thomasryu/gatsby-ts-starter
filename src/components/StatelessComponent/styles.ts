import { css } from 'emotion'

export const componentTitle = css`
  display: block;
  color: red;
  font-size: 32px;
  margin-bottom: 40px;
`
