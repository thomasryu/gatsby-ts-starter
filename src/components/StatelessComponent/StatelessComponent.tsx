import * as React from 'react'
import * as styles from './styles'

interface IProps {
  title: string
}

export const StatelessComponent = (props: IProps) => {
  return (
    <>
      <h1 className={styles.componentTitle}>{props.title}</h1>
    </>
  )
}
