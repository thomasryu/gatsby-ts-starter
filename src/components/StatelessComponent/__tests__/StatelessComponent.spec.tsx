import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { StatelessComponent } from '../StatelessComponent'

Enzyme.configure({ adapter: new Adapter() })

const createStatelessComponent = (title: string) => {
  const testTarget = shallow(<StatelessComponent title={title} />)
  return { testTarget }
}

it('renders the component as expected', () => {
  const setup = createStatelessComponent('Title')
  expect(setup.testTarget).toMatchSnapshot()
})
