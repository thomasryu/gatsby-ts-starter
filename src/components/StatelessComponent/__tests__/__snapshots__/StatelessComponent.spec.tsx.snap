// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StatelessComponent
    title="Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <h1
        className="css-tg3g4y"
      >
        Title
      </h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Title",
        "className": "css-tg3g4y",
      },
      "ref": null,
      "rendered": "Title",
      "type": "h1",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <h1
          className="css-tg3g4y"
        >
          Title
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title",
          "className": "css-tg3g4y",
        },
        "ref": null,
        "rendered": "Title",
        "type": "h1",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
